### PR TITLE
Abductor tools and tool sound refactor.

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Abductor Agents have now been equipped with extremely advanced construction and hacking tools.
/:cl:

The tools:
welder Welding Tool: Faster, doesn't blind and consumes no fuel.
screw Screwdriver: Faster.
crow Crowbar: Faster.
wrench Wrench: Faster.
wire Wirecutters: Faster.
mutli Multitool: Shows the function of the wires when hacking.
belt Belt: A reskinned military belt.

Sprites by me.

Also in this PR:
- A lot of things that used hardcoded sounds now use the tool's usesound.
- Some tool operations properly check for the toolspeed.
- Made alien surgery tools faster. Surgery now checks for toolspeed, so this also affects ghetto tools with toolspeed.
- Adds construction and surgery tool designs in r&d, so they have something if they get abductor tech. Abductor construction tools are only obtainable from the abductor antag and not from the ruin in lavaland.

Reason for the PR: i want to start fleshing out the abductor theme so that it can become more than a one-dimensional side antag. Also, this PR gives actual rewards to researching abductor tech that goes beyond fancy walls and floors.
The tool sound refactor was necessary to enable the custom tool sounds, plus it makes the behaviour consistent across all tools, which can now use different usesounds if needed in future.

### CODE REVIEW

File: code/datums/wires/wires.dm [hunk 1]
```diff
@@ -200,7 +200,7 @@ var/list/wire_name_directory = list()
 	for(var/color in colors)
 		payload.Add(list(list(
 			"color" = color,
-			"wire" = (IsAdminGhost(user) ? get_wire(color) : null),
+			"wire" = (IsAdminGhost(user) || (user.is_holding_item_of_type(/obj/item/device/multitool/abductor)) ? get_wire(color) : null),
```
Comment (ChangelingRain) [inline code/datums/wires/wires.dm]: ewwww

File: code/game/objects/items/weapons/tools.dm [hunk 2]
```diff
@@ -580,6 +605,19 @@
 /obj/item/weapon/weldingtool/mini/flamethrower_screwdriver()
 	return
 
+/obj/item/weapon/weldingtool/abductor
+	name = "alien welding tool"
+	desc = "An alien welding tool. Whatever fuel it uses, it never runs out."
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "welder"
+	toolspeed = 0.1
```
Comment (ChangelingRain) [inline code/game/objects/items/weapons/tools.dm]: This is preeeeetty ridiculously fast.

Comment (XDTM) [inline code/game/objects/items/weapons/tools.dm]: i wanted them to be better than the 0.25 two-in-one roundstart CE tools, but i'll scale them back if you want

File: code/datums/wires/wires.dm (hunk 1 above)
Comment (XDTM) [inline code/datums/wires/wires.dm]: i don't know how to do it on the tool's side :(

File: code/game/objects/items/weapons/tools.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/game/objects/items/weapons/tools.dm]: honestly the ce roundstart tools should be slower, 4X speed is really good on its own; these should probably be like 0.2 or 0.25

Comment (Supermichael777) [inline code/game/objects/items/weapons/tools.dm]: should just have a mini esword

**Comment by coiax** [comment]
The only sprite I have a problem with is the multitool, it's not that recognisable as a multitool, and looks similar to the science tool. Otherwise, I'm not sure why abductors need tools, but provided they have suitable descriptions, then I'm cool with it,

**Comment by XDTM** [comment]
mutli
would this be better?

**Comment by ShadowDeath6** [comment]
do these explode in the face of any non abductor that tries to use it? if yes then :+1:

**Comment by XDTM** [comment]
right now no, but you still have to pry them off an agent's cold, dead hands

File: code/datums/wires/wires.dm (hunk 1 above)
Comment (GunHog) [inline code/datums/wires/wires.dm]: Put var on tool, set it to TRUE for advanced tool, check var here

**Comment by jughu** [comment]
Its not like it would matter abductors having fwst tools since they can teleport everywhere

**Comment by XDTM** [comment]
No, of course, but i want to flesh out the abductor theme since i like it. Plus, crew can loot it and use it.

**Comment by jughu** [comment]
Crew already can get syndiekeys trough abductors fuck off with these loot pinatas

**Comment by XDTM** [comment]
They're just faster tools, from a rarely seen and _very_ slippery antag. Killing a single traitor usually nets you more loot.

**Comment by jughu** [comment]
i mean can we atleast remove the traitor channell off the syndie headset they have or just put a intercom instead of a headset in their dropship they dont need it anyways since they cant talk

**Pull request merged**